### PR TITLE
Purge WRDS API Dependency from Viz Pipelines

### PR DESCRIPTION
**Pull request by shawncrawley**
The changes herein originated from ticket #772 - namely improving the process that synchronizes the on-prem WRDS location DB with our cloud version. The syncing process started failing a couple of months back. Turns out that was due to permission errors on the WRDS-side when trying to write the Location DB dump to S3. So with the help of @DrixTabligan-NOAA, we went ahead and tackled #590 to get the DB dump uploaded to the NWM Shared bucket instead. 

Once #590 was complete, the syncing process continued to fail due to the WRDS API test failing on the new location DB dumps. This is expected behavior, since the most recent changes to the WRDS DB were more significant. However, rather than just fix the failing code, I also completely removed the dependency on the WRDS API (since none of our services depend on it anymore) and replaced it with tests that are DB-centric.

The terraform code that creates the Test WRDS DB lambda packages every SQL file across our whole repository into a folder that the lambda has access to. When the tests run, the code creates a foreign db connection to the `wrds_location3_ondeck` db via a new `test_external` schema and also creates a `automated_test` schema. Then the code iterates through every SQL file and checks if it has any dependency on the WRDS db (i.e. references to the `external` schema). If the dependency exists, the SQL is tweaked on the fly to access the `wrds_location3_ondeck` db via the `test_external` schema, and also to write any intermediate `SELECT ... INTO` results to the `automated_test` db instead. If all of the relevant SQL files execute successfully, the tests pass and the `wrds_lcoation3_ondeck` db is swapped for the live (i.e. non-`_ondeck`) version.

That's that! No more dependence upon the WRDS API and even more robust and relevant testing.

### CODE REVIEW

File: Core/StepFunctions/main.tf [hunk 1]
```diff
@@ -216,13 +225,88 @@ resource "aws_cloudwatch_event_rule" "viz_pipeline_step_function_failure" {
...
+  name     = "hv-vpp-${var.environment}-restore-db-from-s3"
+  role_arn = var.viz_lambda_role
+
+  definition = templatefile("${path.module}/restore_db_from_s3_dump.json.tftpl", {
+    ensure_ec2_ready_step_function_arn  = aws_sfn_state_machine.ensure_ec2_ready_for_use_step_function.arn
+    rds_bastion_id = var.rds_bastion_id
+    region = var.region
+  })
+}
+
```
Comment (shawncrawley) [inline Core/StepFunctions/main.tf:260]: @DrixTabligan-NOAA The terraform that sets up the lambda pass-through event that triggers the sync-wrds-location-db step function can go here.

**Comment by shawncrawley** [comment]
We need to get this in ASAP. The WRDS DB syncing has been failing for months again since the initial changes were put in place manually. A VLab ticket was just opened a few days ago pointing out that our DB is out of sync: https://vlab.noaa.gov/redmine/issues/136423.

@DrixTabligan-NOAA Let's tag up on this.

File: Core/StepFunctions/main.tf (hunk 1 above)
Comment (shawncrawley) [reply Core/StepFunctions/main.tf:261]: We talked and decided to leverage the existing `viz-initialize-pipeline` Lambda function to receive the WRDS DB dump SNS messages and call the `hv-vpp-ti-sync-wrds-location-db` Step Function.

**Review by DrixTabligan-NOAA — APPROVED** [review]
Looks good.

**Pull request merged**